CMakeLists.txt

```diff
@@ -4,8 +4,8 @@ project(gtda_bindings LANGUAGES CXX)
 add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/gtda/externals/pybind11)
 set(BINDINGS_DIR "gtda/externals/bindings")
 
-find_package(Boost 1.56 REQUIRED)
-include_directories(${Boost_INCLUDE_DIR})
+find_package(Boost 1.57 REQUIRED)
```

Comment:
Could you explain the reason for the version bump to 1.57?

Reply:
I changed it to 1.57 after I got the following error message multiple times:

`class boost::optional<Gudhi::rips_complex::Rips_complex<double> >’ has no member named ‘emplace’`

Apparently this emplace method appeared from 1.57 onwards as mentioned here:
https://github.com/p4lang/p4c/issues/1113#issuecomment-353126261

Reply:
I should really prepare a `docker` with all the minimal requirements and see if everything works or something else will break.

Thank you that you spotted that

Reply:
I created a docker with only the minimal requirements and it build without a problem, I went deeper into the boost API.

From the documentation, they introduce emplace with version `>=1.56` [API](https://www.boost.org/doc/libs/1_56_0/libs/optional/doc/html/boost_optional/reference/detailed_semantics.html)
But in version `1.55` the method does not exist [API](https://www.boost.org/doc/libs/1_55_0/libs/optional/doc/html/boost_optional/detailed_semantics.html)

Do you want me to try to build your PR locally with standard `CmakeLists.txt`?

Reply:
@MonkeyBreaker please test this if you can.

Reply:
@ulupo I just tested on the same configuration as my previous test.
I just removed the changes on `CMakeLists.txt`, and installation worked just fine.

Python 3.6.9
Boost: 1.56.0

Reply:
@MonkeyBreaker @gtauzin I have now reverted these changes from this PR as they are a bit out of scope, but we should keep investigating this and if necessary open a new PR with fixes.